File: Python/d1.py
res = 0
aux = 0
with open("inputd1.txt","r") as input:
    for line in input:
        mass = int(line)
        fuel = mass//3 - 2
        res += fuel
        while ((fuel//3)-2) > 0:
            fuel = ((fuel//3)-2)
            aux += fuel
print(res)
print(aux)
print(res+aux)
print("---------------")

File: C++/d1.cpp
int main() {
    string line;
    int res = 0;
    ifstream input("inputd1.txt");
    if(input.is_open()) {
        while(input.is_good()) {
            getline(input, line);
            int mass = atoi(line);
            int fuel = mass / 3;
            if(fuel > 2) res += (fuel -2);
        }
        input.close();
    }
}